File: day1a/js/main.js
import { readFileSync } from "fs";

const FILE_PATH = "../input.txt";

const input = readFileSync(FILE_PATH, { encoding: "utf8" });
const lines = input.split(/\r?\n/);

let runningTotal = 0;
let maxSoFar = 0;
for (const line of lines) {
	if (line === "") {
		if (runningTotal > maxSoFar) {
			maxSoFar = runningTotal;
		}
		runningTotal = 0;
		continue;
	}

	const number = Number.parseInt(line);
	runningTotal += number;
}

console.log(maxSoFar);

File: day1a/go/main.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

const FILE_PATH = "../input.txt"

func main() {
	input, err := os.ReadFile(FILE_PATH)

	if err == nil {
		lines := strings.Split(string(input), "\n")
		running_total := 0
		max_so_far := 0

		for _, line := range lines {
			if line == "" {
				if running_total > max_so_far {
					max_so_far = running_total
				}
				running_total = 0
				continue
			}

			number, err := strconv.ParseInt(line, 10, 0)
			if err == nil {
				running_total += int(number)
			}
		}

		fmt.Println(max_so_far)
	}

}
